pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentStatsHistory.java

```diff
@@ -233,9 +235,21 @@ public synchronized boolean isEmpty() {
     return getNumntriesToScan() == 0;
   }
 
+  public void setMinIntervalBetweenUpdatesMillis(long millis) {
+    _minIntervalBetweenUpdatesMillis = millis;
+  }
+
   public synchronized void addSegmentStats(SegmentStats segmentStats) {
```

Comment:
Also, I noticed that MutableSemgentImpl is the one that calls addSegmentStats followed by a save(). Should save also not happen if we haven't updated due to the interval check? I am wondering if the interface can be simplified to have addSegmentStats call save internally. That way its well contained and we don't have to rely on users calling or not calling save().